images/poetry-python3.11/context/Makefile

```diff
@@ -19,14 +19,14 @@ poetry-generate-requirements: requirements-poetry.in ## Generate requirements-po
 		--verbose \
 		requirements-poetry.in
 
-pip-tools-generate-requirements: requirements-pip-tools.in ## Generate requirements-pip-tools.txt
+pip-tools-generate-requirements: docker-compose/requirements-pip-tools.in ## Generate requirements-pip-tools.txt
 	pip-compile \
 		--allow-unsafe \
 		--annotate \
 		--generate-hashes \
 		--resolver backtracking \
-		--output-file requirements-pip-tools.txt \
+		--output-file docker-compose/requirements-pip-tools.txt \
 		--rebuild \
 		--resolver backtracking \
 		--verbose \
-		requirements-pip-tools.in
+		docker-compose/requirements-pip-tools.in
```

Comment:
A minor optimisation do make things a bit cleaner, by doing this `make` will check the timestamp the prerequisite (`docker-compose/requirements-pip-tools.in`) and the target (`docker-compose/requirements-pip-tools.txt`) and only run the recipe if the prerequisite is newer than the target.

```Makefile
.PHONY: pip-tools-generate-requirements
pip-tools-generate-requirements: docker-compose/requirements-pip-tools.txt

docker-compose/requirements-pip-tools.txt: docker-compose/requirements-pip-tools.in ## Generate requirements-pip-tools.txt
	pip-compile \
		--allow-unsafe \
		--annotate \
		--generate-hashes \
		--resolver backtracking \
		--output-file $@ \
		--rebuild \
		--resolver backtracking \
		--verbose \
		docker-compose/requirements-pip-tools.in
```

Reply:
I am going to do this in a separate PR.